tools/vscode-azurewebpubsub/src/workflows/service/copyEndpoint/CopyEndpointStep.ts

```diff
@@ -0,0 +1,29 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardExecuteStep } from "@microsoft/vscode-azext-utils";
+import * as vscode from 'vscode';
+import { env, type Progress } from "vscode";
+import { type ICopyEndpointContext } from "./ICopyEndpointContext";
+import { localize } from "../../../utils";
+
+export class CopyEndpointStep extends AzureWizardExecuteStep<ICopyEndpointContext> {
+    public priority: number = 110;
+
+    public async execute(context: ICopyEndpointContext, _progress: Progress<{ message?: string | undefined; increment?: number | undefined }>): Promise<void> {
+        if (!context.subscription || !context.serviceName || !context.resourceGroupName || !context.endpoint) {
```

Comment:
since only context.endpoint and context.serviceName are used, is checking these 2 enough?

Reply:
better to check all properties. they're all expected to be existing. And this will help troubleshooting.